File: module4/case_study/furama_resort/src/main/java/com/example/case_study/model/employee/Employee.java
package com.example.case_study.model.employee;

import com.example.case_study.model.contract.Contract;
import com.fasterxml.jackson.annotation.JsonBackReference;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "employee")
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String dateOfBirth;

    private String idCard;

    private double salary;

    private String phoneNumber;

    private String email;

    private String address;

    @ManyToOne
    @JoinColumn(name = "position_id",referencedColumnName = "id")
    private Position position;

    @ManyToOne
    @JoinColumn(name = "education_degree_id",referencedColumnName = "id")
    private EducationDegree educationDegree;

    @ManyToOne
    @JoinColumn(name = "division_id",referencedColumnName = "id")
    private Division division;

    @OneToMany(mappedBy = "employee")
    @JsonBackReference
    private Set<Contract> contracts;

}

File: module5/case-study-rev2/src/app/employee/model/employee.ts
import {Division} from './division';
import {EducationDegree} from './education-degree';

export interface Employee {
  id: number;
  name?: string;
  dateOfBirth?: string;
  idCard?: string;
  salary?: number;
  phoneNumber?: string;
  email?: string;
  address?: string;
  position?: Position;
  division?: Division;
  educationDegree?: EducationDegree;
  flagDelete?: boolean;
}
